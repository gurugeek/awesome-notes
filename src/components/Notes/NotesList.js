import React, { PureComponent } from 'react';
import { View, Text, VirtualizedList } from 'react-native';
import PropTypes from 'prop-types';

class NotesList extends PureComponent {

    _renderItem({ item }) {
        return (
            <View style={{ height: 100, width: '100%', alignItems: 'center', justifyContent: 'center' }}>
                <Text>{item.text}</Text>
            </View>
        );
    }

    _getItem(data, index) {
        return data[index];
    }

    _shouldItemUpdate(prev, next) {
        return prev.item !== next.item;
    }

    render() {
        console.log(this.props.notes);
        return (
            <View style={{ flex: 1 }}>
                <Text>Notes List</Text>
                <VirtualizedList
                    data={this.props.notes}
                    renderItem={this._renderItem}
                    shouldItemUpdate={this._shouldItemUpdate}
                    windowSize={10}
                    keyExtractor={(item, index) => item.id}
                    getItemCount={(data) => data.length}
                    getItem={this._getItem}
                    showsVerticalScrollIndicator={false}
                />
            </View>
        );
    }
}

NotesList.propTypes = {
    notes: PropTypes.array.isRequired
};

export default NotesList;