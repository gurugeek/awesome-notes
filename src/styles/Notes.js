export default {
    container: {
        flex: 1,
        backgroundColor: '#1D3557',
        padding: 10
    },
    emptyNotes: {
        wrapper: {
            flex: 1,
            backgroundColor: '#457B9D',
            borderRadius: 5,
            justifyContent: 'center',
            padding: 10
        },
        iconRow: {
            marginBottom: 10,
            alignItems: 'center'
        },
        titleRow: {
            marginBottom: 40
        },
        titleText: {
            backgroundColor: 'transparent',
            fontFamily: 'Roboto-Bold',
            fontSize: 24,
            color: '#F1FAEE',
            textAlign: 'center'
        },
        messageRow: {
            marginBottom: 20
        },
        messageText: {
            backgroundColor: 'transparent',
            fontFamily: 'Roboto-Regular',
            fontSize: 18,
            color: '#F1FAEE',
            textAlign: 'center'
        },
        gradient: {
            colors: ['#e4efe9', '#93a5cf'],
            wrapper: {
                paddingVertical: 10,
                borderRadius: 5
            }
        },
        buttonText: {
            textAlign: 'center',
            backgroundColor: 'transparent',
            fontFamily: 'Roboto-Bold',
            fontSize: 18,
            color: '#1D3557'
        }
    },
    notesList: {
        wrapper: {
            flex: 1
        }
    },
    noteItem: {
        swipeoutWrapper: {
            backgroundColor: 'transparent',
            marginBottom: 10
        },
        wrapper: {
            backgroundColor: '#A8DADC',
            borderRadius: 10,
            padding: 10
        },
        textWrapper: {
            flex: 1,
            marginBottom: 5
        },
        text: {
            color: '#211A1E',
            fontFamily: 'Roboto-Regular'
        },
        dateWrapper: {
            height: 10
        },
        date: {
            color: '#457B9D',
            fontSize: 10,
            textAlign: 'right',
            fontFamily: 'Roboto-Thin'
        }
    },
    noteButton: {
        wrapper: {
            height: '100%',
            alignItems: 'center',
            justifyContent: 'center',
            backgroundColor: 'transparent'
        }
    }
}