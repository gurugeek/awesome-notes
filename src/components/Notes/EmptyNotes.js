import React from 'react';
import {View, Text, TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';

import { NotesStyle } from '../../styles/index';

const EmptyNotes = ({ gotoCreateNote }) => {
    return (
        <View style={NotesStyle.emptyNotes.wrapper}>
            <View style={NotesStyle.emptyNotes.iconRow}>
                <Icon name="ios-book-outline" size={50} color="white" />
            </View>
            <View style={NotesStyle.emptyNotes.titleRow}>
                <Text style={NotesStyle.emptyNotes.titleText}>You have no notes...yet...</Text>
            </View>
            <View style={NotesStyle.emptyNotes.messageRow}>
                <Text style={NotesStyle.emptyNotes.messageText}>There are no notes saved. Create your first one by tapping on the following button!</Text>
            </View>
            <TouchableOpacity onPress={gotoCreateNote}>
                <LinearGradient locations={[0.1,1.0]} colors={NotesStyle.emptyNotes.gradient.colors} style={NotesStyle.emptyNotes.gradient.wrapper}>
                    <Text style={NotesStyle.emptyNotes.buttonText}>Create new note!</Text>
                </LinearGradient>
            </TouchableOpacity>
        </View>
    );
};

EmptyNotes.propTypes = {
    gotoCreateNote: PropTypes.func.isRequired
};

export default EmptyNotes;