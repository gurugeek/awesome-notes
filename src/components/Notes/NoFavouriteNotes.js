import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import { NotesStyle } from '../../styles/index';

const NoFavouriteNotes = () => {
  return (
    <View style={NotesStyle.emptyNotes.wrapper}>
        <View style={NotesStyle.emptyNotes.iconRow}>
            <Icon name="ios-star-outline" size={50} color="#FDE74C" />
        </View>
        <View style={NotesStyle.emptyNotes.titleRow}>
            <Text style={NotesStyle.emptyNotes.titleText}>You have no favourite notes..yet...</Text>
        </View>
        <View style={NotesStyle.emptyNotes.messageRow}>
            <Text style={NotesStyle.emptyNotes.messageText}>Add your first one swiping your note from left to right and tapping on the <Icon name="ios-star-outline" size={16} color="#FDE74C" /></Text>
        </View>
    </View>
  );
};

export default NoFavouriteNotes;