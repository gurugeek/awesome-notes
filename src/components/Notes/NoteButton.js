import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Ionicons';

import { ICON_TYPES } from '../../constants/IconTypes';
import  { NotesStyle } from '../../styles';

const NoteButton = ({ type }) => {
    return (
        <View style={NotesStyle.noteButton.wrapper}>
            <Icon name={ICON_TYPES[type].name} size={30} color={ICON_TYPES[type].color} />
        </View>
    );
};

NoteButton.propTypes = {
    type: PropTypes.string.isRequired
};

export default NoteButton;