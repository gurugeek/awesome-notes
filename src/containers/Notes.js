import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import EmptyNotes from '../components/Notes/EmptyNotes';

class Notes extends Component {

    static navigationOptions = ({ navigation }) => ({
        title: 'Notes',
    });

    render() {
        return (
            <View style={{ flex: 1, backgroundColor: '#1D3557', padding: 10 }}>
                <EmptyNotes addNote={() => { console.log("add note");}} />
            </View>
        );
    }
}

export default Notes;