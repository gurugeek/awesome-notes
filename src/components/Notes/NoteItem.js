import React, { Component } from 'react';
import { View, Text, Animated } from 'react-native';
import PropTypes from 'prop-types';
import Swipeout from 'react-native-swipeout';

import { NotesStyle } from '../../styles';
import NoteButton from './NoteButton';

class NoteItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            scaleAnim: new Animated.Value(1),
            opacityAnim: new Animated.Value(1)
        };

        this._switchFavorite = this._switchFavorite.bind(this);
        this._onSwipeOpen = this._onSwipeOpen.bind(this);
        this._onSwipeClose = this._onSwipeClose.bind(this);
    }

    _runScaleAnimation(toValue) {
        Animated.timing(this.state.scaleAnim, { toValue }).start();
    }

    _runOpacityAnimation(toValue) {
        Animated.timing(this.state.opacityAnim, { toValue }).start();
    }

    _formatDate(date) {
        const [day, month, year] = ['' + date.getDate(), '' + (date.getMonth() + 1), date.getFullYear()];
        return [day, month, year].join('/');
    }

    _switchFavorite() {
        this.props.note.favourite = !this.props.note.favourite;
        this.props.updateNote(this.props.note);
    }

    _onSwipeOpen(sectionID, rowId, direction) {
        this._runScaleAnimation(0.60);
        direction == 'right' && this._runOpacityAnimation(0.5);
    }

    _onSwipeClose(sectionID, rowId, direction) {
        this._runScaleAnimation(1);
        direction === 'right' && this._runOpacityAnimation(1);
    }

    render() {
        const swipeSettings = {
            autoClose: true,
            right: [{ onPress: () => { this.props.deleteNote(this.props.note.id)}, component: <NoteButton type='delete' />, backgroundColor: 'transparent' }],
            left: [{ onPress: this._switchFavorite, component: <NoteButton type={this.props.note.favourite ? 'favourite-selected' : 'favourite'} />, backgroundColor: 'transparent'}],
            rowId: this.props.note.id,
            onOpen: this._onSwipeOpen,
            onClose: this._onSwipeClose,
            style: NotesStyle.noteItem.swipeoutWrapper
        };

        let { scaleAnim, opacityAnim } = this.state;
        const rotateY = this.state.opacityAnim.interpolate({
            inputRange: [0.5, 1],
            outputRange: ['45deg', '0deg']
        });
        return (
            <Swipeout {...swipeSettings}>
                <Animated.View style={[NotesStyle.noteItem.wrapper, { opacity: opacityAnim }, { transform: [{ scale: scaleAnim }, { perspective: 1000 }, { rotateY }] }]}>
                    <View style={NotesStyle.noteItem.textWrapper}>
                        <Text style={NotesStyle.noteItem.text}>{this.props.note.text}</Text>
                    </View>
                    <View style={NotesStyle.noteItem.dateWrapper}>
                        <Text style={NotesStyle.noteItem.date}>{this._formatDate(this.props.note.createdAt)}</Text>
                    </View>
                </Animated.View>
            </Swipeout>
        );
    }
}

NoteItem.propTypes = {
    note: PropTypes.object.isRequired,
    updateNote: PropTypes.func.isRequired,
    deleteNote: PropTypes.func.isRequired
};

export default NoteItem;