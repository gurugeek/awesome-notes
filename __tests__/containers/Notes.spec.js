import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Notes from '../../src/containers/Notes';

Enzyme.configure({ adapter: new Adapter() });

const storeFake = {
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {
        return {
            notes: {
                filterByFavourites: false
            }
        };
    },
};

const mockNavigation = {
    setParams: () => {}
};

describe('<Notes /> container', () => {

    it('Notes should match with the snapshot', () => {
        const tree = renderer.create(
            <Provider store={storeFake}>
                <Notes navigation={mockNavigation} />
            </Provider>
        );
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const notesShallow = shallow(
        <Provider store={storeFake}>
            <Notes navigation={mockNavigation} />
        </Provider>
    );

    it('Notes should exists', () => {
        expect(notesShallow.length).toEqual(1);
    });
});