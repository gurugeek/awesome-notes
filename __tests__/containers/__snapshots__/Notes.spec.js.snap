// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notes /> container Notes should match with the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1D3557",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#457B9D",
        "borderRadius": 5,
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 50,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        Ôèß
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#F1FAEE",
            "fontFamily": "Roboto-Bold",
            "fontSize": 24,
            "textAlign": "center",
          }
        }
      >
        You have no notes...yet...
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#F1FAEE",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        There are no notes saved. Create your first one by tapping on the following button!
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <BVLinearGradient
        colors={
          Array [
            4293193705,
            4287866319,
          ]
        }
        endPoint={undefined}
        locations={
          Array [
            0.1,
            1,
          ]
        }
        startPoint={undefined}
        style={
          Object {
            "borderRadius": 5,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1D3557",
              "fontFamily": "Roboto-Bold",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Create new note!
        </Text>
      </BVLinearGradient>
    </View>
  </View>
</View>
`;
